### ISSUE TITLE
pynn.random blows up if you dont set a parameters parameter 

### ISSUE THREAD

**Issue by alan-stokes** [NeuralEnsemble/PyNN#438]
this came about due to us in spinnaker trying to test a pynn 0.8 impl of our tools. We had a virtual environment that had pynn 0.8 installed via pip. When we accidentally ran a pynn 0.75 script with this virtual environment, we hit a issue in pynn.random when our possion was trying to get some data. 

After a bit of debugging, I noticed that the error is internal to the pynn.random file, and so it should be possible to recreate it with a pynn 0.8 script, as long as your don't set the parameters parameter to the next() function (this has not been tested, but should be correct from below deductions).

Line 109 of the random.py (next() function) creates a set of 2 floats when the parameters parameter is not set. The parameters is then used on line 202 (_next() function), which asks for its keys property. As a set doesn't have keys, it crashes. 

as there are no docs of the next() or _next() function's parameters, I don't know what it expects parameters to be, and so cant recommend a fix.

hope this helps 
Alan

**Comment by apdavison** [comment]
ok, I can reproduce this:
```python
In [1]: from pyNN.random import NumpyRNG

In [2]: rng = NumpyRNG(seed=493875632)

In [3]: rng.next()
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-3-16c08bba24ff> in <module>()
----> 1 rng.next()

/home/docker/dev/PyNN/pyNN/random.py in next(self, n, distribution, parameters, mask_local)
    111             rarr = numpy.random.rand(0)  # We return an empty array
    112         elif n is None:
--> 113             rarr = self._next(distribution, 1, parameters)
    114         elif n > 0:
    115             if self.num_processes > 1 and not self.parallel_safe:

/home/docker/dev/PyNN/pyNN/random.py in _next(self, distribution, n, parameters)
    200         # TODO: allow non-standardized distributions to pass through without translation
    201         distribution_np, parameter_map = self.translations[distribution]
--> 202         if set(parameters.keys()) != set(parameter_map.keys()):
    203             # all parameters must be provided. We do not provide default values (this can be discussed).
    204             errmsg = "Incorrect parameterization of random distribution. Expected %s, got %s."

AttributeError: 'tuple' object has no attribute 'keys'
```

**Comment by apdavison** [comment]
Fixed in 128a53d
